core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -668,7 +668,7 @@ public DiscoveryNodes build() {
             ImmutableOpenMap.Builder<String, DiscoveryNode> masterNodesBuilder = ImmutableOpenMap.builder();
             ImmutableOpenMap.Builder<String, DiscoveryNode> ingestNodesBuilder = ImmutableOpenMap.builder();
             Version minNodeVersion = Version.CURRENT;
-            Version maxNodeVersion = Version.CURRENT;
+            Version maxNodeVersion = null;
```

Comment:
yeah, that has to do with the fact that an empty builder (which can't really exist, but that's not clear from this code) will report min/max as CURRENT. This means that you can init the min to the highest possible value (current) but the max can't as CURRENT is higher than anything else you might encounter in the nodes. I'll just change the code to consistent and avoid confusion.